File: RepoTransBench/target_projects/Java/Go/Hitomis_CircleMenu/tests/original/on_menu_selected_listener_test.go
package original

import (
	"testing"
)

type OnMenuSelectedListener interface {
	OnMenuSelected(index int)
}

type MenuSelectedListener struct {
	LastSelectedIndex int
}

func (l *MenuSelectedListener) OnMenuSelected(index int) {
	l.LastSelectedIndex = index
}

func TestOnMenuSelectedCalled(t *testing.T) {
	listener := &MenuSelectedListener{LastSelectedIndex: -1}
	listener.OnMenuSelected(2)
	if listener.LastSelectedIndex != 2 {
		t.Errorf("Expected LastSelectedIndex to be 2, got %d", listener.LastSelectedIndex)
	}
	listener.OnMenuSelected(0)
	if listener.LastSelectedIndex != 0 {
		t.Errorf("Expected LastSelectedIndex to be 0, got %d", listener.LastSelectedIndex)
	}
}

File: RepoTransBench/target_projects/Java/C#/Hitomis_CircleMenu/tests/original/OnMenuSelectedListenerTest.cs
using Xunit;

namespace Hitomis_CircleMenu.Tests
{
    public class OnMenuSelectedListenerTest
    {
        class TestListener : IOnMenuSelectedListener
        {
            public int LastSelectedIndex = -1;
            public void OnMenuSelected(int index)
            {
                LastSelectedIndex = index;
            }
        }

        [Fact]
        public void TestOnMenuSelectedCalled()
        {
            var listener = new TestListener();
            listener.OnMenuSelected(2);
            Assert.Equal(2, listener.LastSelectedIndex);
            listener.OnMenuSelected(0);
            Assert.Equal(0, listener.LastSelectedIndex);
        }
    }

    public interface IOnMenuSelectedListener
    {
        void OnMenuSelected(int index);
    }
}